src/common/device_helpers.cuh

```diff
@@ -391,6 +391,24 @@ using device_vector = thrust::device_vector<T,  XGBDeviceAllocator<T>>;  // NOLI
 template <typename T>
 using caching_device_vector = thrust::device_vector<T,  XGBCachingDeviceAllocator<T>>;  // NOLINT
 
+// Faster to instantiate than caching_device_vector and invokes no synchronisation
+// Use this where vector functionality (e.g. resize) is not required
+template <typename T>
+class TemporaryArray {
```

Comment:
Maybe `FixedSizeArray` ...

Reply:
Array already implies a fixed size: https://en.cppreference.com/w/cpp/container/array

Reply:
That's compile time.

Reply:
Nonetheless I would like to keep the name. This also follows thrust::detail::temporary_array.